### PR TITLE
V30 admin improvements

### PR DESCRIPTION
**Pull request by cdosborn**
## Description

There are several improvements to the admin experience included here. 

- On resource request approval the reason field is omitted, which makes much more sense in the email template
- Identities can be patched (to update quota)
- RT no longer receives requests (there is now one queue to review resource requests the atmosphere admin, this needs to be approved by @edwins)

See the tropo pr: https://github.com/cyverse/troposphere/pull/728

## Checklist before merging Pull Requests
- [ ] New test(s) included to reproduce the bug/verify the feature
- [ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature
- [ ] If creating/modifying DB models which will contain secrets or sensitive information, PR to [clank](https://github.com/cyverse/clank) updating sanitation queries in `roles/sanitary-sql-access/templates/sanitize-dump.sh.j2`
- [ ] Reviewed and approved by at least one other contributor.
- [ ] If necessary, include a snippet in CHANGELOG.md
- [ ] New variables supported in Clank
- [ ] New variables committed to secrets repos

### CODE REVIEW

File: api/permissions.py [hunk 1]
```diff
@@ -271,6 +271,28 @@ def has_permission(self, request, view):
...
+
+        method_is_safe = request.method in permissions.SAFE_METHODS
+        no_bad_params = \
+            not any([p for p in ["all_users", "username"] if p in request.GET])
+
+        # For all other users, only allow SAFE_METHODS and for GET don't allow
+        # filtering by all_users or username
+        return (
+            method_is_safe and no_bad_params
+        )
```
Comment (steve-gregory) [inline api/permissions.py]: Nice use of permissions 👍  I appreciate the "human language" for variable names.

File: api/tests/v2/test_resource_requests.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        resource_request = create_request_for_user(user)
+
+    url = reverse(view_path, args=[resource_request.pk])
+    view = viewset.as_view({'patch': 'partial_update'})
+
+    request = APIRequestFactory().patch(url, payload, format='json')
+    force_authenticate(request, user=user)
+
+    response = view(request, pk=resource_request.pk)
+    return response
```
Comment (steve-gregory) [inline api/tests/v2/test_resource_requests.py]: This method could probably be used elsewhere... But looks good for now 👍

File: api/v2/serializers/details/identity.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 
 class IdentitySerializer(serializers.HyperlinkedModelSerializer):
-    quota = QuotaSummarySerializer(source='get_quota')
+    quota = QuotaWritableSummarySerializer()
```
Comment (steve-gregory) [inline api/v2/serializers/details/identity.py]: Could you explain a bit about why you went this route over `ModelRelatedField`?

File: api/v2/serializers/details/resource_request.py [hunk 4]
```diff
@@ -13,8 +14,12 @@ class ResourceRequestSerializer(serializers.HyperlinkedModelSerializer):
         view_name='api:v2:resourcerequest-detail',
     )
     created_by = UserSummarySerializer(read_only=True)
-    identity = IdentityRelatedField(source='membership.identity')
-    status = StatusTypeSummarySerializer(read_only=True)
+    status = ModelRelatedField(
+        default=lambda: get_status_type(status="pending"),
```
Comment (steve-gregory) [inline api/v2/serializers/details/resource_request.py]: Fancy defaults you got there 😉

File: api/v2/serializers/summaries/quota.py [hunk 5]
```diff
@@ -20,3 +21,23 @@ class Meta:
...
+    def to_internal_value(self, data):
+        try:
+            pk = data["id"]
+            return self.get_queryset().get(pk=pk)
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError("Object does not exist for id: %s" % pk)
+        except (KeyError, TypeError, ValueError):
+            raise serializers.ValidationError(
+                "Expected dictionary with primary key like: { id: <Int> }. "
+                "Intead got: %s" % str(data))
```
Comment (steve-gregory) [inline api/v2/serializers/summaries/quota.py]: I believe this is pretty similar to `ModelRelatedField`? [Source](https://github.com/cyverse/atmosphere/blob/master/core/serializers/fields/base.py#L40-L46)

File: api/v2/views/identity.py [hunk 6]
```diff
@@ -23,14 +23,14 @@ class Meta:
         fields = ["project_id", "project_uuid", "group_id"]
 
 
-class IdentityViewSet(MultipleFieldLookup, AuthModelViewSet):
+class IdentityViewSet(MultipleFieldLookup, UserListAdminQueryAndUpdate, ModelViewSet):
```
Comment (steve-gregory) [inline api/v2/views/identity.py]: Are there other `ViewSet` classes that use this behavior (Users read, admins can update)?

If so, we could package that up similar to what we've done with `AuthModelViewSet`..

File: api/v2/views/resource_request.py [hunk 7]
```diff
@@ -1,43 +1,32 @@
...
-        else:
-            queryset = self.model.objects.filter(created_by=self.request.user)
-
-        # Note:
-        # Select_related is better but can only be used with 1-1, foreign key
-        # Prefetch_related is required for M2M
-        # See https://stackoverflow.com/a/31237071/1213041
-        return queryset \
+        return \
+            self.model.objects.all() \
```
Comment (steve-gregory) [inline api/v2/views/resource_request.py]: So simple! ❤️

File: core/migrations/0098_remove_resourcerequest_membership.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    dependencies = [
+        ('core', '0097_userprofile_guacamole_color'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='resourcerequest',
+            name='membership',
+        ),
+    ]
```
Comment (steve-gregory) [inline core/migrations/0098_remove_resourcerequest_membership.py]: Reminder to whomever merges the PR that migrations are included, this is the third `0098` active PR 😉

File: core/models/resource_request.py [hunk 9]
```diff
@@ -1,18 +1,26 @@
...
 from django.db import models
+from django.utils import timezone
+from core.models.user import AtmosphereUser
+from core.models.status_type import StatusType
 
-from core.models.abstract import BaseRequest
 
-
-class ResourceRequest(BaseRequest):
+class ResourceRequest(models.Model):
```
Comment (steve-gregory) [inline core/models/resource_request.py]: I'm for removing `BaseRequest` altogether in a separate PR...

Comment (cdosborn) [reply core/models/resource_request.py]: I think BaseRequest could make sense if membership gets removed. I think this model could become the new BaseRequest, with a refactor.

File: api/v2/views/identity.py (hunk 6 above)
Comment (cdosborn) [reply api/v2/views/identity.py]: Yes, but its not clear to me whether custom perm like this are better than having separate views. I think I like the latter more.

File: core/templates/core/email/resource_request_approved.html [hunk 10]
```diff
@@ -7,8 +7,6 @@
 
 Your resource request for {{request}} has been approved!
 
-Reason: {{reason}}
-
```
Comment (cdosborn) [inline core/templates/core/email/resource_request_approved.html]: This email really didn't make sense before, and most staff weren't aware of what they were sending.

File: core/migrations/0098_remove_resourcerequest_membership.py (hunk 8 above)
Comment (julianpistorius) [reply core/migrations/0098_remove_resourcerequest_membership.py]: Oh boy. Thanks for the warning. :)

**Pull request merged**